### PR TITLE
http: allowing the codec to access stream info

### PR DESCRIPTION
**Pull request by alyssawilk**
This allows Envoy the mobile codec impl to have access to stream info

Risk Level: n/a (extending a C++ interface)
Testing: unit tested
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Review by goaway — APPROVED** [review]
Thank you @alyssawilk! This looks great, and I believe it gives us access to everything we need on the library/mobile side to surface error callbacks as needed.

File: source/common/http/conn_manager_impl.h [hunk 1]
```diff
@@ -191,6 +184,16 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     // Http::RequestDecoder
     void decodeHeaders(RequestHeaderMapPtr&& headers, bool end_stream) override;
     void decodeTrailers(RequestTrailerMapPtr&& trailers) override;
+    const StreamInfo::StreamInfo& streamInfo() const override {
+      return filter_manager_.streamInfo();
```
Comment (mattklein123) [inline source/common/http/conn_manager_impl.h:188]: When is this allocated? Can this ever be null between returning the decoder to the codec and this being called?

**Review by mattklein123 — COMMENTED** [review]
LGTM with one small question, thanks.

/wait-any

File: source/common/http/conn_manager_impl.h (hunk 1 above)
Comment (alyssawilk) [reply source/common/http/conn_manager_impl.h:188]: stream info is a member variable of filter_manager_ and initialized on construction:
https://github.com/envoyproxy/envoy/blob/main/source/common/http/filter_manager.h#L989

Ditto filter_manager_ and ActiveStream
https://github.com/envoyproxy/envoy/blob/main/source/common/http/conn_manager_impl.h#L351

Comment (mattklein123) [reply source/common/http/conn_manager_impl.h:188]: OK thank for confirming.

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**